documentation/milestones/individual_milestones/get_opportunities.md

```diff
@@ -0,0 +1,192 @@
+  - changes made to a given endpoint to be backward-compatible, so that I can start building against this API without worrying about breaking changes
+- As a **project maintainer**, I want:
+  - most of the critical development infrastructure to be in place when we officially launch the API, so that we can deploy bug fixes or new features quickly once the site is live
+- As an **open source contributor**, I want:
+  - to be able run the code for the API locally, so that I can test my contributions to the codebase
+
+## Technical description
+
+### Infrastructural Requirements
+
+The infrastructure developed to deploy and host the API should balance:
+
+- Code quality
+- Security
+- Delivery velocity
+- Cost & maintenance
+
+### Developer Experience Requirements
+
+The API should be developed and versioned in a way that balances:
+
+- Discoverability
+- Ease of adoption
+- Backwards compatibility
+
+### Data Requirements
+
+The way that data is stored and delivered through the API should balance:
+
+- Eventual consistency with legacy Grants.gov
+- Improvements to the existing data model
+- Ease of managing schema changes
+
+### Definition of done
+<!-- Required -->
+
+- [ ] The following infrastructure requirements are satisfied:
+  - [ ] The code needed to build and deploy the site is merged to `main`
+  - [ ] The site is built and hosted with the tools selected in the [API Planning](https://github.com/HHS/grants-equity/issues/42) and [DB Planning](https://github.com/HHS/grants-equity/issues/48) milestones
+  - [ ] All code quality checks set up in the [Developer Tools milestone](https://github.com/HHS/grants-equity/issues/50) are passing
+  - [ ] The resources required to deploy and host the API are provisioned programmatically using the [Infrastructure-as-Code milestone](https://github.com/HHS/grants-equity/issues/123) framework
+  - [ ] Code changes are deployed using the CI/CD pipeline set up in [the Back-end CI/CD milestone](https://github.com/HHS/grants-equity/issues/57)
+- [ ] The following developer experience (DX) requirements are satisfied:
+  - [ ] The API is live at `beta.grants.gov/api/`
+  - [ ] Developers can learn how to interact with the API by referencing the API documentation
+  - [ ] The endpoint path indicates which major version of the API the developer is consuming
+  - [ ] Breaking changes are reflected in predictable changes to the endpoint path
+  - [ ] The endpoint is available when legacy grants.gov experiences planned or unplanned downtime, maintenance, and upgrades
+- [ ] The following data requirements are satisfied:
+  - [ ] The endpoint returns all of the grant opportunities that are available on grants.gov
+  - [ ] The endpoint returns at least one (1) field per opportunity
+  - [ ] Updates to the data in legacy Grants.gov are propagated to the new endpoint within 24 hours
+
```

Comment:
- Do we want to set up any monitoring/alerting? It seems that we'll have logging set up with the metrics we've defined below but do we want to add monitoring/alerting if there is a downtime? 
- Do we want to perform load testing to assess how the API performs under heavy user traffic?

Reply:
> Do we want to set up any monitoring/alerting? It seems that we'll have logging set up with the metrics we've defined below but do we want to add monitoring/alerting if there is a downtime?

Good call! I added a bullet for logging in [this commit](https://github.com/HHS/grants-equity/pull/249/commits/ea40a70d48173e6a2399793eb63b8b2620016c1d) 

**Follow up question:** Do you think it would be helpful to track logging/monitoring as a 10k ft body of work (similar to Backend CI/CD)?

> Do we want to perform load testing to assess how the API performs under heavy user traffic?

I had initially scoped performance and load testing under the Search API/UI milestone, thinking that it might be hard to fit into this milestone, but agree that performance testing will be critical before we launch the search functionality and could be useful even at this stage. 

**Follow up question:** Do you think it would be worth pulling the performance testing framework body of work up into this milestone?

Reply:
1. Yeah, that makes sense that logging/monitoring and alerting be its own body of work. 
2. I assume we're probably still keeping the traffic fairly minimal when we release this, yes? I lean towards conducting smaller scoped standalone performance testing under this milestone. This might inform our later Search API/UI milestone where we can establish a more comprehensive performance testing framework. Not sure if that makes technical sense, but think i's important to do the initial performance testing so we have a smooth first release, but still give the team time to establish an intentional performance testing framework later for larger scale traffic.

Reply:
I like the idea of working to setup performance testing now. Just a reminder: even if some definition-of-done requirement (such as perf testing) is part of the milestone, we can still deploy the API to PROD before all the definitions of done are complete. So we could deploy the APIs to PROD, then later run performance testing. 

Adding things as part of the definition of done for the milestone just means we can't mark the milestone complete until those things are done. 

As for alerting, we have this milestone in the short descriptions: https://github.com/HHS/grants-equity/blob/main/documentation/milestones/milestone_short_descriptions.md?plain=1#L44. @widal001 , it's looking like it actually did not get created as a GitHub Projects ticket -- do you want to do that? Incident response is mentioned in this ticket: https://github.com/HHS/grants-equity/issues/131, but that's only a planning ticket, not an implementation ticket. 

It would make sense to me to get that setup as part of this milestone, although maybe like perf testing, it's something that happens after launch.

Reply:
Thanks for this discussion!! Based on this thread, I added performance testing to the list of dependencies and referenced it in the infrastructure definition of done in [this commit](https://github.com/HHS/grants-equity/pull/249/commits/3aa14fa418605012f4e9766952271ca22b97e8f2).